### PR TITLE
fix and modify the GithubProfileScraper.java

### PR DESCRIPTION
**Pull request by vibhcool**
At present, GithubProfileScraper.java, is not able to scrap all data. 

 - This pr targets to make it workable and scrap the data 
 - This pr adds new data field userId to be scraped

### CODE REVIEW

File: src/org/loklak/api/search/GithubProfileScraper.java [hunk 1]
```diff
@@ -70,10 +72,12 @@ public JSONObject serviceImpl(Query call, HttpServletResponse response, Authoriz
 	public static SusiThought scrapeGithub(String profile) {
 
 		Document html = null;
-
+        String user_id;
```
Comment (yukiisbored) [inline src/org/loklak/api/search/GithubProfileScraper.java]: Indentation inconsistency

File: src/org/loklak/api/search/GithubProfileScraper.java [hunk 2]
```diff
@@ -70,10 +72,12 @@ public JSONObject serviceImpl(Query call, HttpServletResponse response, Authoriz
...
+    
 		JSONObject githubProfile = new JSONObject();
 
 		try {
+            /* Use scraper */
```
Comment (yukiisbored) [inline src/org/loklak/api/search/GithubProfileScraper.java]: Indentation inconsistency

File: src/org/loklak/api/search/GithubProfileScraper.java [hunk 3]
```diff
@@ -100,56 +104,72 @@ public static SusiThought scrapeGithub(String profile) {
 			json.setData(arr);
 			return json;
 		}
+        
+
+		String avatarUrl = html.getElementsByAttributeValueContaining("class", "avatar").attr("src");
 
-		String avatarUrl = html.getElementsByAttributeValue("class", "avatar rounded-2").attr("src");
-		githubProfile.put("avatar_url", avatarUrl);		
+        Pattern avatar_url_to_user_id = Pattern.compile(".com\\/u\\/([0-9]+)\\?");
```
Comment (yukiisbored) [inline src/org/loklak/api/search/GithubProfileScraper.java]: Indentation inconsistency

File: src/org/loklak/api/search/GithubProfileScraper.java [hunk 4]
```diff
@@ -100,56 +104,72 @@ public static SusiThought scrapeGithub(String profile) {
...
 		githubProfile.put("works_for", worksFor);
 
-		String homeLocation = html.getElementsByAttributeValue("itemprop", "homeLocation").attr("title");
+		String homeLocation = html.getElementsByAttributeValueContaining("itemprop", "homeLocation").attr("title");
+        
 		githubProfile.put("home_location", homeLocation);
 
-		String email = html.getElementsByAttributeValue("itemprop", "email").text();
+		String email = html.getElementsByAttributeValueContaining("itemprop", "email").text();
+        if (!email.contains("@"))
```
Comment (yukiisbored) [inline src/org/loklak/api/search/GithubProfileScraper.java]: Indentation inconsistency

File: src/org/loklak/api/search/GithubProfileScraper.java [hunk 5]
```diff
@@ -100,56 +104,72 @@ public static SusiThought scrapeGithub(String profile) {
...
 
-		String specialLink = html.getElementsByAttributeValue("itemprop", "url").text();
+		String specialLink = html.getElementsByAttributeValueContaining("itemprop", "url").text();
 		githubProfile.put("special_link", specialLink);
 
-		String joiningDate = html.getElementsByAttributeValue("class", "join-date").attr("datetime");
-		githubProfile.put("joining_date", joiningDate);
-
+		Elements joiningDates = html.getElementsByAttributeValueContaining("class", "dropdown-item");
+        for(Element joiningDate: joiningDates) {
```
Comment (yukiisbored) [inline src/org/loklak/api/search/GithubProfileScraper.java]: Indentation inconsistency

**Comment by daminisatya** [comment]
@vibhcool Can you please attach screenshot or provide a gist link of the final JSON output?

**Comment by vibhcool** [comment]
@yukiisbored , fixed indentation errors

**Comment by vibhcool** [comment]
@daminisatya 
screenshot from 2017-04-21 11-38-54

**Comment by vibhcool** [comment]
@daminisatya ^^ screenshot of the output of json

**Comment by vibhcool** [comment]
testted, open for review :)

**Comment by vibhcool** [comment]
@yukiisbored  thank you :)

**Pull request merged**